src/lib/uTop.ml

```diff
@@ -288,9 +288,32 @@ let parse_default parse str eos_is_error =
       | Syntaxerr.Ill_formed_ast (loc, s) ->
           Error ([mkloc loc],
                  Printf.sprintf "Error: broken invariant in parsetree: %s" s)
-      | Syntaxerr.Invalid_package_type (loc, s) ->
-          Error ([mkloc loc],
-                 Printf.sprintf "Invalid package type: %s" s)
+      | Syntaxerr.Invalid_package_type (loc, err) ->
+#if OCAML_VERSION >= (5, 2, 0)
```

Comment:
Can you move that block to `UTop_compat` as a `invalid_package_error_to_string`? This module has no `mli` file so it can take a string in <5.2.0 and the variant in >=5.2.0.